src/classes/CCON_PreventUpdate_TEST.cls

```diff
@@ -0,0 +1,207 @@
+        Course_offering__c newCourseOffering = UTIL_UnitTestData_TEST.createCourseOffering(NULL, NULL);
+        
+        course_Enrollment__c courseEnrollment = [SELECT Id, Contact__c, Course_Offering__c FROM Course_Enrollment__c LIMIT 1];
+        courseEnrollment.Contact__c = newCon.Id;
+        courseEnrollment.Course_offering__c = newCourseOffering.Id;
+        Test.startTest();
+            update courseEnrollment;
+        Test.stopTest();    
+        
+        System.assertEquals(courseEnrollment.Contact__c, newCon.Id);
+        System.assertEquals(courseEnrollment.Course_offering__c, newCourseOffering.Id);
+    
+    }
+    
+    // Test Contact and Course Offering updates with Term Grades related to the Course Connection 
+    
+    @isTest 
+    static void testUpdateCourseConnWithTermGrade() {
+    
+        course_Enrollment__c courseEnrollment = [SELECT Id, Contact__c, Course_Offering__c FROM Course_Enrollment__c LIMIT 1];
+        course_offering__c courseOffering = [SELECT Id FROM Course_Offering__c LIMIT 1];
+        Contact con = [SELECT Id FROM Contact LIMIT 1];
+        courseEnrollment.Course_Offering__c = courseOffering.Id;
+        update courseEnrollment;
+        
+        // Insert Term Grades
+        Term_Grade__c termGrade = new Term_Grade__c(Course_Connection__c = courseEnrollment.Id, Course_Offering__c = courseOffering.Id, Contact__c = con.Id);
+        insert termGrade;
+        
+        //Insert a contact
+        Contact newCon = New Contact();
+        newCon.FirstName = 'TestNew02';
+        newCon.LastName = 'LastNew02';
+        newCon.Email = 'Last02@sfdo.com';
+        newCon.Phone = '7676876788';
+        insert newCon;
+        
+        // Insert a Course Offering
+        Course_offering__c newCourseOffering = UTIL_UnitTestData_TEST.createCourseOffering(NULL, NULL);
+        
+        courseEnrollment.Course_Offering__c = newCourseOffering.Id;
+        courseEnrollment.Contact__c = newCon.Id;
+        try {
+            Test.startTest();
+                update courseEnrollment;
+            Test.stopTest();    
+        } catch (Exception error) {
+            System.assert(error.getMessage().contains(Label.CannotChangeContactAndCourseOffering));      
+        }
+    }
+    
+    // Test Contact and Course Offering updates with Attendance Events related to the Course Connection 
+    
+    @isTest 
+    static void testUpdateCourseConnWithAttendance() {
+    
+        course_Enrollment__c courseEnrollment = [SELECT Id, Contact__c, Course_Offering__c FROM Course_Enrollment__c LIMIT 1];
+        course_offering__c courseOffering = [SELECT Id FROM Course_Offering__c LIMIT 1];
+        Contact con = [SELECT Id FROM Contact LIMIT 1];
+        courseEnrollment.Course_Offering__c = courseOffering.Id;
+        update courseEnrollment;
+        
+        // Insert Attendance Event
+        Attendance_Event__c AttendanceEvent = UTIL_UnitTestData_TEST.getAttendanceEvent();
```

Comment:
Could the UTIL_UnitTestData_TEST class method be updated to accept additional parameters?

Reply:
These are used in other test classes as well, changing this will require updates to old ones. I think this was purposely created for some reason this way, not sure though.

Reply:
Can always add a new method that takes in additional parameters - so it would look something like

```
private static Attendance_Event__c getAttendanceEvent() {
    return new Attendance_Event__c();
}

private static Attendance_Event__c getAttendanceEvent(param1, param2, ...) {
    Attendance_Event__c attendanceEvent = getAttendanceEvent();
    attendanceEvent.Some_Field__c = param1;
    // ...
    return attendanceEvent;
}
```